climate-advisor/vector_db/utils/text_processing.py

```diff
@@ -0,0 +1,242 @@
+"""
+Text processing utilities for document splitting and embedding preparation.
+
+This module provides functions for:
+- PDF text extraction
+- Text chunking/splitting
+- Text cleaning and preprocessing
+"""
+
+import re
+from pathlib import Path
+from typing import List, Optional, Dict, Any
+from uuid import uuid4
+
+from PyPDF2 import PdfReader
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+
+
+class PDFProcessor:
+    """Handles PDF text extraction and processing."""
+
+    @staticmethod
+    def extract_text_from_pdf(file_path: str) -> str:
+        """
+        Extract text content from a PDF file.
+
+        Args:
+            file_path: Path to the PDF file
+
+        Returns:
+            Extracted text content as a string
+
+        Raises:
+            FileNotFoundError: If the PDF file doesn't exist
+            Exception: If there's an error reading the PDF
+        """
+        if not Path(file_path).exists():
+            raise FileNotFoundError(f"PDF file not found: {file_path}")
+
+        try:
+            reader = PdfReader(file_path)
+            text_content = []
+
+            for page in reader.pages:
+                text = page.extract_text()
+                if text.strip():  # Only add non-empty pages
+                    text_content.append(text.strip())
+
+            return "\n\n".join(text_content)
```

Comment:
Can we do this once and output the text content to a text file that is added to the repository instead of having the raw (and quite large) text file and processing it each time?